File: backend/src/models/Artist.js
// models/Artist.js
const mongoose = require( 'mongoose' );

// Define a schema with strict mode set to false to allow any fields
const ArtistSchema = new mongoose.Schema( {}, { strict: false } );

// Add an index on 'genres.name'
ArtistSchema.index( { 'genres.name': 1 } );

// Create and export the model
module.exports = mongoose.model( 'Artist', ArtistSchema, 'artists' );

File: backend/src/models/mongoengine/Artist.py
# Artist.py

from mongoengine import (
    DynamicDocument
)
from models.mongoengine.BaseSchema import BaseSchema

class Artist(BaseSchema, DynamicDocument):
    meta = {
        'collection': 'artists',
        'indexes': [
            {'fields': ['genres.name']}
        ]
    }
